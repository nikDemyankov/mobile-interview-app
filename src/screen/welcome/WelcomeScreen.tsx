import React, { memo, useCallback } from 'react';
import { Text, View } from 'react-native';
import LottieView from 'lottie-react-native';

import { Button, Screen } from '../../component';
import styles from './WelcomeScreen.styles';
import { useNavigation } from '../../hooks';

const LurkingCat = require('./resources/lurking-cat.json');

const WelcomeScreen: React.FC = () => {
  const navigation = useNavigation();

  const onPress = useCallback(() => {
    navigation.openInstrumentList();
  }, [navigation]);

  return (
    <Screen>
      <View style={styles.container}>
        <View>
          <Text style={styles.title}>Nordnet Mob App</Text>
          <Text style={styles.paragraph}>
            We are going to do the task as a team, there is no pressure on finishing the task, that
            is not what is important. We are just going to work together for a while.
          </Text>
          <Text style={styles.relax}>Relax and enjoy!</Text>
        </View>
        <LottieView source={LurkingCat} autoPlay loop style={styles.lurkingCat} />
        <View style={styles.button}>
          <Button onPress={onPress} label="Lets get started!" />
        </View>
      </View>
    </Screen>
  );
};

export default memo(WelcomeScreen);
