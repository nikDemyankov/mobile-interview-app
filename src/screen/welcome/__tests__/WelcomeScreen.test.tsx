import React from 'react';
import { act, create } from 'react-test-renderer';
import { Button } from '../../../component';

import WelcomeScreen from '../WelcomeScreen';

const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: mockNavigate,
  }),
}));

jest.mock('lottie-react-native', () => 'LottieView');
jest.mock('../resources/lurking-cat.json', () => 'lurking-cat.json');
jest.mock('../../../component/Button', () => 'Button');
jest.mock('../../../component/Screen', () => 'Screen');

it('should render properly', () => {
  const component = create(<WelcomeScreen />);
  expect(component).toMatchSnapshot();
});

it('should navigate to instrument list', () => {
  const component = create(<WelcomeScreen />);

  expect(mockNavigate).not.toHaveBeenCalled();

  act(() => {
    component.root.findByType(Button).props.onPress();
  });

  expect(mockNavigate).toHaveBeenCalledTimes(1);
  expect(mockNavigate).toHaveBeenCalledWith('InstrumentList', undefined);
});
