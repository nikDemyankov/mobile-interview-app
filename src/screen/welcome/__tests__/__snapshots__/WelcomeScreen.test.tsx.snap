// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<Screen>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "space-between",
        "paddingVertical": 24,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 32,
            "fontWeight": "bold",
            "marginBottom": 16,
            "paddingHorizontal": 32,
            "textAlign": "center",
          }
        }
      >
        Nordnet Mob App
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "lineHeight": 24,
            "paddingHorizontal": 32,
            "textAlign": "center",
          }
        }
      >
        We are going to do the task as a team, there is no pressure on finishing the task, that is not what is important. We are just going to work together for a while.
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 24,
            "marginTop": 12,
            "paddingHorizontal": 32,
            "textAlign": "center",
          }
        }
      >
        Relax and enjoy!
      </Text>
    </View>
    <LottieView
      autoPlay={true}
      loop={true}
      source="lurking-cat.json"
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Button
        label="Lets get started!"
        onPress={[Function]}
      />
    </View>
  </View>
</Screen>
`;
